### PR TITLE
GC events loss detection

### PR DESCRIPTION
**Pull request by sergey-miryanov**

### CODE REVIEW

**Comment by sergey-miryanov** [comment]
@codspeedbot why this regression?

File: docs/formats.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ Perfetto features:
 - **Process command lines**: With the [`[cmdline]` extra](rss.md#the-cmdline-extra), each monitored process's command line is written to the trace — see [Process command lines](#process-command-lines) below.
 - **`Start Process` marker**: A zero-duration instant event named `Start Process` is emitted on each process track at that process's first event. Perfetto hides a track that carries no events, so this guarantees the process track and its label always render. It is Perfetto-only; consumers that enumerate slices should filter it out.
 - **RSS counter track**: A process-level `rss` counter track appears for each PID when `--rss` is enabled, showing Resident Set Size in bytes. Sampled at the configured `--rss-interval` (default 1s).
+- **`GC Loss` track**: One row per interpreter, named `GC Loss {iid}`, sitting under that process's own track. Each slice marks one poll interval in which the target collected without gcmon reading the records. See [GC Loss slices](#gc-loss-slices) below.
```
Comment (sergey-miryanov) [inline docs/formats.md]:
```suggestion
- **`GC Loss` track**: One row per interpreter, named `GC Loss {iid}`, sitting under that process's own track. Each slice marks one poll interval in which gcmon missing one or more records from the target. See [GC Loss slices](#gc-loss-slices) below.
```

File: docs/README.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ how to read the result.
 
 | Page | What it covers |
 |---|---|
+| [monitoring.md](monitoring.md) | How gcmon reads a process, why it misses collections, and what it recovers — the basis for the pages below |
```
Comment (sergey-miryanov) [inline docs/README.md]:
```suggestion
| [monitoring.md](monitoring.md) | How gcmon reads a process, why it misses collections, and what it recovers |
```

**Comment by sergey-miryanov** [comment]
@codspeedbot why this regression?

**Comment by sergey-miryanov** [comment]
@codspeedbot we still have regression

File: src/gcmon/monitor.py [hunk 3]
```diff
@@ -82,61 +98,84 @@ def poll(self, pid: int) -> PollStatus:
...
+            if not unseen:
                 continue
-            if event.collections <= cursors.get((event.iid, event.gen), 0):
+
+            gen_loss = accumulator.ingest(unseen)
+            gens_by_iid.setdefault(iid, []).append(gen_loss)
+            self._stats.record_lifetime(pid, iid, gen, accumulator.last_collections, accumulator.last_duration)
+            if gen_loss.lost_count:
+                self._stats.record_loss(pid, gen, gen_loss.lost_count, gen_loss.lost_pause_ns)
+            fresh.extend(unseen)
```
Comment (maurycy) [inline src/gcmon/monitor.py:147]: @sergey-miryanov 

I'm wondering if this "catch up" is OK when monitoring is started, paused, then started again.

Comment (sergey-miryanov) [reply src/gcmon/monitor.py:147]: Ah, good catch! I completely forgot that we can pause monitoring :)

The more I think about the `pyperf` hook, the more convinced I am that we should just emit instant events for benchmark entry and exit, rather than starting/stopping monitoring. In fact, it might even be worth removing the start/stop/pause commands from the control plane entirely.

Comment (maurycy) [reply src/gcmon/monitor.py:147]: @sergey-miryanov 

I agree that start / stop / pause is hard. ;-)

I think this guarantee might be too strong:

https://github.com/sergey-miryanov/gcmon/pull/86/changes#diff-a58f64d699a4475cb378bc0d60d80565f85d23e4682c6ed615fe56fce598cadaR127

If it's easened, I think that `gcmon` could get away with two internal counters (ie: how many times it read, and how many times it's started.)
